src/compas_timber/elements/fasteners/fastener.py

```diff
@@ -0,0 +1,165 @@
+    """
+    A class to represent timber fasteners (screws, dowels, brackets).
+
+    Parameters
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    Attributes
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    """
+
+    def __init__(self, elements, **kwargs):
+        super(Fastener, self).__init__(elements, **kwargs)
+        self.elements = elements
+        self.features = []
+        self.attributes = {}
+        self.attributes.update(kwargs)
+        self.debug_info = []
+
+    def __repr__(self):
+        # type: () -> str
+        element_str = ["{} {}".format(element.__class__.__name__, element.key) for element in self.elements]
+        return "Fastener({})".format(", ".join(element_str))
+
+    # ==========================================================================
+    # Computed attributes
+    # ==========================================================================
+
+    @property
+    def is_fastener(self):
+        return True
+
+
+    @property
+    def shape(self):
+        # type: () -> Brep
+        return self._create_shape(self.frame, self.beams)
+
+    @property
+    def key(self):
+        # type: () -> int | None
+        return self.graph_node
+
+    def __str__(self):
+        element_str = ["{} {}".format(element.__class__.__name__, element.key) for element in self.elements]
+        return "Fastener connecting {}".format(", ".join(element_str))
+
+    # ==========================================================================
+    # Implementations of abstract methods
+    # ==========================================================================
+
+    def compute_geometry(self):
+        # type: (bool) -> compas.geometry.Brep
+        """Compute the geometry of the fastener.
+
+        Returns
+        -------
+        :class:`compas.geometry.Brep`
+
+        """
+        raise NotImplementedError
+
+    def compute_aabb(self, inflate=0.0):
+        # type: (float) -> compas.geometry.Box
+        """Computes the Axis Aligned Bounding Box (AABB) of the element.
+
+        Parameters
+        ----------
+        inflate : float, optional
+            Offset of box to avoid floating point errors.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Box`
+            The AABB of the element.
+
+        """
+        raise NotImplementedError
+
+    def compute_obb(self, inflate=0.0):
+        # type: (float | None) -> compas.geometry.Box
+        """Computes the Oriented Bounding Box (OBB) of the element.
+
+        Parameters
+        ----------
+        inflate : float
+            Offset of box to avoid floating point errors.
+
+        Returns
+        -------
+        :class:`compas.geometry.Box`
+            The OBB of the element.
+
+        """
+        raise NotImplementedError
```

Comment:
shouldn't these be implemented? if not then there's no reason to re-define these, they are defined in `Element` already.